src/classes/frPledge.cls

```diff
@@ -0,0 +1,88 @@
+/*
+*
+*  Copyright (c) 2020, Funraise Inc
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions are met:
+*  1. Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*  2. Redistributions in binary form must reproduce the above copyright
+*     notice, this list of conditions and the following disclaimer in the
+*     documentation and/or other materials provided with the distribution.
+*  3. All advertising materials mentioning features or use of this software
+*     must display the following acknowledgement:
+*     This product includes software developed by the <organization>.
+*  4. Neither the name of the <organization> nor the
+*     names of its contributors may be used to endorse or promote products
+*     derived from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY FUNRAISE INC ''AS IS'' AND ANY
+*  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+*  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL FUNRAISE INC BE LIABLE FOR ANY
+*  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+*  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+*  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+*  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+*  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+*
+*
+* PURPOSE:  A class for representing a Funraise pledge.  Used for creating & matching to pledges
+*
+*
+*
+* CREATED: 2020 Funraise Inc - https://funraise.io
+* AUTHOR: Alex Molina
+*/
+public class frPledge {
+    public static Pledge__c findActive(Id contactId) {
+        List<Pledge__c> activePledges = [SELECT Id FROM Pledge__c 
+                                         WHERE Supporter__c = :contactId 
+                                         AND (Start_Date__c = null OR Start_Date__c >= TODAY)
+                                         AND (End_Date__c = null OR End_Date__c <= TODAY)
+                                         AND Active__c = true
+                                         AND Pledge_Donation__c = null
+                                         ORDER BY CreatedDate ASC LIMIT 1
```

Comment:
This query should also check to make sure the funraise__Percent_Complete__c formula field is less than 100, since we don't want to consider pledges that have reached their pledge amount as active